### ISSUE TITLE
Wrong/surprising result from noncommutative Pow.subs

### ISSUE THREAD

**Issue by pv** [sympy/sympy#13004]
The following behavior in Sympy seems surprising:
```
>>> import sympy
>>> sympy.__version__
'1.1'
>>> x = sympy.Symbol('x', commutative=False)
>>> (x*x*x).subs({x*x: 1})   # !!!
1
```
I would have expected this produces `x`.

The issue appears to be that `Pow._eval_subs` uses fractional powers in the substitution. This then raises questions on what noncommutative symbols in Sympy actually mean mathematically. At least in my use cases for them, certainly `x**2 == 1` does not imply `x**3 == 1`.

I would suggest something like the following:
```
--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -591,6 +591,11 @@ def _check(ct1, ct2, old):
         if old == self.base:
             return new**self.exp._subs(old, new)

+        if not old.is_commutative:
+            # The logic here does not work for noncommutative objects.
+            # x*x == 1 should not imply x == 1.
+            return False, None
+
         # issue 10829: (4**x - 3*y + 2).subs(2**x, y) -> y**2 - 3*y + 2
         if old.func is self.func and self.exp == old.exp:
             l = log(self.base, old.base)
```

**Comment by asmeurer** [comment]
Agreed, fractional power logic should not apply to noncommutatives. Does your patch result in `x` or `x**3` for the substitution?

**Comment by pv** [comment]
It results to `x**3`, so there's room for improvement.

**Comment by asmeurer** [comment]
If you're happy with `x**3` you can use `xreplace` instead of `subs` as a workaround. `xreplace` just does exact substitution, no mathematical manipulation.

**Comment by pv** [comment]
Somewhat better fix in #13018.
